abyss/django/middleware.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+import shutil
+from tempfile import SpooledTemporaryFile
+
+from django.core.files.storage import get_storage_class
+from django.utils.module_loading import import_string
+
+from abyss.django.settings import (
+    ABYSS_DJANGO_BUFFER_SIZE,
+    ABYSS_DJANGO_FILE_PREFIX,
+    ABYSS_DJANGO_REQUEST_FILTER,
+    ABYSS_DJANGO_STORAGE,
+)
+from abyss.simple import profiling
+
+
+class AbyssMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.request_filter = import_string(ABYSS_DJANGO_REQUEST_FILTER)
```

Comment:
I'm a tiny bit on the fence about the import_string indirection here – it feels like an end user who needs more control over the request filter could subclass AbyssMiddleware in their project and use that (since changing the request filter will necessarily need code changes anyway, not just configuration).

On the other hand, maybe at that point all of these settings and overrides should be class-level variables for the middleware, á la

```
class AbyssMiddleware:
    buffer_size = 1048576
    file_prefix = "abyss-django/"
    
    ...
    
    def should_profile_request(self, request) -> bool:
        ...  # default implementation

    def get_storage(self) -> Storage:
        ...  # probably default django storage
```

and one could then subclass that for their project?

Reply:
I suppose it's about whether or not we'd want to keep multiple options around or not. Personally I'm not opposed to requiring subclassing, but that could end up being unnecessary boilerplate? Then again you'd have to define a new function somewhere regardless, at which point subclassing would about as much effort required.

So sure, let's change to subclassing instead of string imports